### ISSUE TITLE
Genie fails to start

### ISSUE THREAD

**Issue by luciodaou** [arkane-systems/genie#122]
Hello, I'm facing something similar to https://github.com/arkane-systems/genie/issues/93 on v1.35 and Ubuntu 20.10 on WSL2.

Initialization doesn't finish. If Ctrl+C is pressed and a terminal open, trying the following commands to access the bottle doesn't work and WSL2 becomes extremely slow, with lots of memory usage. Even `wsl.exe -t <distro>` doesn't kill it, `vmcompute.exe` must be killed through Task Manager of taskkill.exe on an elevated prompt.

```
❯ genie -v -s
genie: outside bottle, systemd pid: 3189.
genie: starting shell
Failed to get shell PTY: Connection timed out
genie: starting shell failed; machinectl shell returned 1.
lucio in ~
❯ genie -v -i
genie: outside bottle, systemd pid: 3189.
genie: bottle already exists (no need to initialize).
lucio in ~
❯ genie -v -u
genie: outside bottle, systemd pid: 3189.
genie: running systemctl poweroff within bottle
Failed to set wall message, ignoring: Connection timed out
Failed to power off system via logind: Connection timed out
Failed to start poweroff.target: Connection timed out
See system logs and 'systemctl status poweroff.target' for details.
It is possible to perform action directly, see discussion of --force --force in man:systemctl(1).
genie: running command failed; nsenter returned 1.
lucio in ~
❯
```

**Comment by luciodaou** [comment]
I managed to get things working by applying the following commands before starting systemd-genie:
```
sudo systemctl disable ssh.service multipathd.service multipathd.socket
sudo systemctl mask systemd-remount-fs.service
```
I applied both and worked, but reading the comments from where they're from, maybe just the last is needed. Links below:
https://gist.github.com/djfdyuruiry/6720faa3f9fc59bfdf6284ee1f41f950#gistcomment-3631203
https://gist.github.com/djfdyuruiry/6720faa3f9fc59bfdf6284ee1f41f950#gistcomment-3642102
(It's a script to install older versions of genie and dotnet-runtime-3.1 on Ubuntu 20.04)

Now `genie -i` takes a few seconds and everything works fine and flawlessly. Tested on 2 different PCs, both with Ubuntu 20.10.

**Comment by luciodaou** [comment]
On Ubuntu 20.04, initialization took a looooooong time but worked. Then I was able to `systemctl list-units` and disable the ones in magenta. On the following initialization, it was faster, but not even close to as fast as it's now on 20.10 with the commands mentioned above applied.

**Comment by cerebrate** [comment]
Yeah, as you found out, what you've got is a variant of #110 rather than #93. I can't repro your specific variant, though: I just sideloaded Ubuntu 20.10 and it fails for me in the same way as 20.04.

Added a specific note to the README pointing Ubuntu users at #110.

**Comment by PavelSosin-320** [comment]
@cerebrate genie fails to start 1st time because of lazy systemd-networkd-wait-online.service

systemd status shows degraded
and journalctl shows
A start job for unit systemd-networkd-wait-online.service has finished with a failure 
and systemctl restart systemd-networkd-wait-online.service
takes long time and ends with 
the same
and journalctl -xeu systemd-networkd-wait-online.service shows
A start job for unit systemd-networkd-wait-online.service has finished with a failure.
...................
The issue is in the Windows 10. WiFi initialization in Windows 10 is slow that I can watch exclamation sign on the top of WiFi icon during few minutes after any reboot or reconfig. I tried to disable systemd-networkd-wait-online but everything is frozen until Network Manager is completed. There is recommendation fromMS to disable WiFi powesaving device mode but then the list of  the available networks went out of control.
This is the price to pay for keeping WSL network on the air by systemd Network manager.  😒

**Comment by cerebrate** [comment]
@PavelSosin-320 

Doesn't repro for me. For one thing, if your WiFi initialization is taking that long in Windows 10, something is wrong with your WiFi setup; mine is practically instantaneous.

Also, NetworkManager isn't the same thing as systemd-networkd; and by default, the former isn't installed on WSL Ubuntu distros, and the latter isn't configured to manage the eth0 interface, so neither should cause issues to appear on a vanilla genie install. I'm the last person to criticize someone for wanting to push the boundaries of what WSL is capable of, but unconventional network configurations and their problems are out of scope for genie.

**Comment by PavelSosin-320** [comment]
@cerebrate and others: This is 100% systemd issue :  systemd-networkd-wait-online service has many problems and wrong timeout is one of them. Disable it systemctl disable systemd-networkd-wait-online  and the distro will start in seconds and systemctl is-system-running will show running (!!!) instead of degraded.
networkctl will display eth0 as routable. resolvectl query --cache=false works correctly and uses eth0 to resolve domain names.

**Comment by PavelSosin-320** [comment]
I achieved it in a very simple way: Journalctl
shows systemd-networkd-wait-online.service failure during boot and its
restart fails too. It hangs foreaware because of multiple systemd issues
with this service. I disabled it - *systemctl disable
systemd-networkd-wait-online.service* It doesn't harm any WSL networking
functionality.

**Comment by cerebrate** [comment]
From a vanilla Ubuntu 20.10 test install:

```
root@athena-wsl:~# systemctl status systemd-networkd-wait-online
● systemd-networkd-wait-online.service - Wait for Network to be Configured
     Loaded: loaded (/lib/systemd/system/systemd-networkd-wait-online.service; enabled; vendor preset: disabled)
     Active: active (exited) since Fri 2021-03-12 00:47:40 CST; 3min 8s ago
       Docs: man:systemd-networkd-wait-online.service(8)
    Process: 675 ExecStart=/lib/systemd/systemd-networkd-wait-online (code=exited, status=0/SUCCESS)
   Main PID: 675 (code=exited, status=0/SUCCESS)

Mar 12 00:47:40 athena-wsl systemd[1]: Starting Wait for Network to be Configured...
Mar 12 00:47:40 athena-wsl systemd[1]: Finished Wait for Network to be Configured.
root@athena-wsl:~#
```

The service works fine.
